kotlin-testing-2/src/test/kotlin/com/baeldung/mockk/MatchVarargsUnitTest.kt

```diff
@@ -0,0 +1,137 @@
+package com.baeldung.mockk
+
+import io.mockk.MockKException
+import io.mockk.every
+import io.mockk.mockk
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.api.assertThrows
+import kotlin.test.assertEquals
+import kotlin.test.assertTrue
+
+class MyClass {
+    fun joinBySpace(vararg strings: String): String {
+        return strings.joinToString(separator = " ")
+    }
+}
+
+class MatchVarargsUnitTest {
+    private val mockkObj = mockk<MyClass>()
+
+    @Test
+    fun `when calling concatStringsBySpace() on the real object then get expected result`() {
+        val result = MyClass().joinBySpace("a", "b", "c", "d", "e")
+        assertEquals("a b c d e", result)
+    }
+
+    @Test
+    fun `when using anyVararg() on the mocked object then get expected result`() {
+        every { mockkObj.joinBySpace("a", "b", "c", *anyVararg(), "x", "y", "z") } returns "Wow, Kotlin rocks!"
+
+        val result = mockkObj.joinBySpace("a", "b", "c", "Baeldung", "is", "cool", "x", "y", "z")
+        assertEquals("Wow, Kotlin rocks!", result)
+
+        val result2 = mockkObj.joinBySpace("a", "b", "c", "x", "y", "z")
+        assertEquals("Wow, Kotlin rocks!", result2)
+    }
+
+
+    @Test
+    fun `when using varargAll() on the mocked object then get expected result`() {
+        every { mockkObj.joinBySpace("a", "b", "c", *varargAll { it.startsWith("d") }, "z") } returns "Wow, Kotlin rocks!"
```

Comment:
What happens in the "zero arg" case for varaargAll?

Reply:
addressed